storage/remote/intern.go

```diff
@@ -48,7 +47,11 @@ func newEntry(s string) *entry {
 	return &entry{s: s}
 }
 
-func newPool() *pool {
+func newPool(reg prometheus.Registerer) *pool {
+	if reg != nil {
+		reg.MustRegister(noReferenceReleases)
```

Comment:
Does this need to also get unregistered at some point?

Reply:
Maybe here we don't need to worry. We already have metrics being registered in `func NewWriteStorage` with the custom registry and do not bother about unregistering. We only have one storage, so this should be fine for now.